borg/helpers.py

```diff
@@ -151,7 +152,13 @@ def load(cls, repository, key=None, force_tam_not_required=False):
     def write(self):
         if self.key.tam_required:
             self.config[b'tam_required'] = True
-        self.timestamp = datetime.utcnow().isoformat()
+        # self.timestamp needs to be strictly monotonically increasing. Clocks often are not set correctly
+        if self.timestamp is None:
+            self.timestamp = datetime.utcnow().isoformat()
+        else:
+            prev_ts = datetime.strptime(self.timestamp, "%Y-%m-%dT%H:%M:%S.%f")
+            incremented = (prev_ts + timedelta(microseconds=1)).isoformat()
+            self.timestamp = max(incremented, datetime.utcnow().isoformat())
```

Comment:
a bit unpretty: once there is a wrong far-future ts in the manifest, it will never ever get correct again, because it will always choose to increment by 1us.